cdap-data-fabric/src/main/java/co/cask/cdap/data/format/RecordFormats.java

```diff
@@ -26,29 +30,56 @@
+      .put(Formats.STRING, SingleStringRecordFormat.class)
+      .put(Formats.CSV, DelimitedStringsRecordFormat.class)
+      .put(Formats.TSV, DelimitedStringsRecordFormat.class)
+      .build();
+  private static final Map<String, Map<String, String>> NAME_SETTINGS_MAP =
+    ImmutableMap.<String, Map<String, String>>builder()
+      .put(Formats.CSV, ImmutableMap.of(DelimitedStringsRecordFormat.DELIMITER, ","))
+      .put(Formats.TSV, ImmutableMap.of(DelimitedStringsRecordFormat.DELIMITER, "\t"))
       .build();
 
   /**
-   * Create a record format for the given format name. Names are first checked against standard names like "CSV" or
-   * "TSV". If they are not a standard name, they are interpreted as fully qualified class names.
+   * Create an initialized record format for the given format specification. The name in the specification is
+   * first checked against standard names like "CSV" or "TSV". If it is a standard name, the corresponding
+   * format will be created, with specification settings applied on top of default settings.
+   * For example, "CSV" will map to the {@link DelimitedStringsRecordFormat}, with a comma as the delimiter,
+   * whereas "TSV" will map to the {@link DelimitedStringsRecordFormat}, with a tab as the delimiter.
+   * If the name is not a standard name, it is interpreted as a class name.
    *
-   * @param name Name of the format to instantiate.
-   * @param <FROM> Type of underlying object the format reads.
-   * @param <TO> Type of object the format reads the underlying object into.
-   * @return Instantiated {@link RecordFormat} based on the given name.
-   * @throws IllegalAccessException if there was an illegal access when instantiating the record format.
-   * @throws InstantiationException if there was an exception instantiating the record format.
-   * @throws ClassNotFoundException if the record format class could not be found.
+   * @param spec the specification for the format to create and initialize
+   * @param <FROM> Type of underlying object the format reads
+   * @param <TO> Type of object the format reads the underlying object into
+   * @return Initialized {@link RecordFormat} based on the given name
+   * @throws IllegalAccessException if there was an illegal access when instantiating the record format
+   * @throws InstantiationException if there was an exception instantiating the record format
+   * @throws ClassNotFoundException if the record format class could not be found
+   * @throws UnsupportedTypeException if the specification is not supported by the format
    */
-  public static <FROM, TO> RecordFormat<FROM, TO> create(String name)
-    throws IllegalAccessException, InstantiationException, ClassNotFoundException {
-    Class<? extends RecordFormat> formatClass = NAME_MAP.get(name);
-    return (RecordFormat<FROM, TO>) (formatClass == null ?
+  public static <FROM, TO> RecordFormat<FROM, TO> createInitializedFormat(FormatSpecification spec)
```

Comment:
Would be good to have some basic test coverage of this method, maybe as a simple addition to DelimitedStringsRecordFormatTest